File: 3000/LC 2212.go
func maximumBobPoints(numArrows int, aliceArrows []int) []int {
    n := len(aliceArrows)
    maxScore := 0
    ans := make([]int, n)
    for i := 1; i < (1 << n); i++ {
        score_Bob := 0
        cur := 0
        tmp := make([]int, n)
        for j, x := range aliceArrows {
            if i & (1 << j) != 0 {
                cur += x + 1
                tmp[j] = x + 1
                score_Bob += j
            } 
        }
        if cur <= numArrows && score_Bob > maxScore {
            tmp[0] += numArrows - cur
            maxScore = score_Bob
            copy(ans, tmp)
        }
    }

    return ans
}

File: 3000/LC 2212.rs
impl Solution {
    pub fn maximum_bob_points(num_arrows: i32, alice_arrows: Vec<i32>) -> Vec<i32> {
        let n = alice_arrows.len();
        let mut max_score = 0;
        let mut ans = vec![0;n];

        for i in 1..(1<<n) {
            let mut tmp = vec![0;n];
            let mut score_Bob = 0;
            let mut cur = 0;
            for (j, &x) in alice_arrows.iter().enumerate() {
                if i & (1 << j) != 0 {
                    cur += x + 1;
                    tmp[j] += x + 1;
                    score_Bob += j as i32;
                }
            }

            if cur <= num_arrows && score_Bob > max_score {
                max_score = score_Bob;
                tmp[0] += num_arrows - cur;
                ans.copy_from_slice(&tmp);
            }
        }

        ans
    }
}